### PR TITLE
For #481: Created BulletBarrage card

### LINKED ISSUE #481: Soldier Card - Implement Bullet Barrage
Size: 1 unit.

Should happen after EnigmaDragons/ProjectNeon#355

### PR DESCRIPTION
**Pull request by paulodamaso**
Closes #481 
- Implemented  `BulletBarrage`  cards by effect composition

### CODE REVIEW

**Comment by paulodamaso** [comment]
@SilasReinagel Sorry,  forgot to wire the `Repeat` effect

**Comment by paulodamaso** [comment]
@SilasReinagel Done, please take a look

File: src/ProjectNeon/Assets/Features/Effects/RepeatOnPrimaryResource.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿using UnityEngine;
+using System.Collections;
+
+/**
+ * Apply an effect while there is enough Primary Resource
+ */
+public class RepeatOnPrimaryResource : Effect
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/Effects/RepeatOnPrimaryResource.cs]: The name is unclear when reading the card in the editor. Maybe name this `RepeatUntilCannotAffordPrimaryResourceCost` or something that expresses the terminating condition.

File: src/ProjectNeon/Assets/Features/Effects/RepeatOnPrimaryResource.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    private Effect _effect;
+
+    public RepeatOnPrimaryResource(Effect effect)
+    {
+        _effect = effect;
+    }
+
+    void Effect.Apply(Member source, Target target)
+    {
+        while (source.State[source.State.ResourceTypes[0]]  > 0)
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/Effects/RepeatOnPrimaryResource.cs]: Having more than 0 doesn't guarantee that the effect can be afforded. Should be ensuring that it has enough to pay for the recurring effect.

Also, this double-indexing means that we need a more convenient getter on `MemberState`.

Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/Effects/RepeatOnPrimaryResource.cs]: This bug might be worth creating a test for, to ensure that it doesn't over-expend or get any discounts on repetitions.

**Pull request merged**